### ISSUE TITLE
error [pimatic-openweather]: Unexpected token i

### ISSUE THREAD

**Issue by lcfm1** [josecastroleon/pimatic-openweather#9]

**Comment by sweetpi** [comment]
Do you have an stack trace (debug message following to the error)?

**Comment by lcfm1** [comment]
How i can make that? these error i saw few times.

**Comment by coccyx00** [comment]
i have the same. error shown:

```
debug [pimatic-openweather]: SyntaxError: Unexpected token I at Object.parse (native) at IncomingMessage.<anonymous> (/home/pi/pimatic-app/node_modules/pimatic-openweather/node_modules/openweathermap/openweather.js:146:21) at IncomingMessage.EventEmitter.emit (events.js:117:20) at _stream_readable.js:920:16 at process._tickCallback (node.js:415:13)
```

**Comment by christophfroehlich** [comment]
same here (0.8.15)
since a few days, this plugin is spamming the logfile with
`[pimatic-openweather] error: Unexpected token I`
and it doesn't retrieve the weather data for hours.. any chance to get rid of this?

**Comment by mwittig** [comment]
Yes, same here. The openweather service is a pain in the a..

I'll look into this tomorrow and will report back about my findings.

**Comment by renearts** [comment]
Same here indeed. When I surf to http://api.openweathermap.org/data/2.5/weather?q=Volkel,nl, I get as response: Invalid API key. Please see http://openweathermap.org/faq#error401 for more info.
The 401 error says I would need an API key. I guess I/we requested data to many times?

**Comment by mwittig** [comment]
"For FOSS developers: we welcome free software and are willing to help you. If you want to use OWM data in your free software application please register an API key and file a ticket describing your application and API key registered. OWM will review your request and grant immunity from usage limit overflow blocking." Besides this, it is possible to acquire a free key for personal use.

Maybe, the blocking is temporarilly only as the error has occurred multiple times in the past. May be it is better to ask inividuals to acquire a ApiKey for their personal use. 

@sweetpi What's your opinion on this? Does it make sense to request an Apikey for pimatic? I am happy to take the initiative.

**Comment by sweetpi** [comment]
An apikey for pimatic would be cool. However we should be sure to meet their licensing terms. I'm not 100% sure, but probably we need to mention openweather somewhere in the ui. I don't know if the readme is enough. http://openweathermap.org/price @mwittig Would you request one from them and ask if mentioning openweather in the plugin readme is enough?

**Comment by sweetpi** [comment]
In any case: We should not repeat the error message and parse the result as an error message, if json parse fails...

**Comment by mwittig** [comment]
@sweetpi OK, I am ready to do this. I have already sorted the error message on my working copy and will add the apikey as an optional feature to your fork of "node-openweathermap". EDIT: Perhaps you give me access to the repo, or I'll do a fork/PR, otherwise.

**Comment by mwittig** [comment]
Published pimatic-openweather@0.8.16

**Comment by poeml** [comment]
Big thank you from my side.

Viele Grüße, Peter Pöml
